src/main/java/com/meta/cp4m/message/ThreadState.java

```diff
@@ -56,16 +60,20 @@ public static <T extends Message> ThreadState<T> of(T message) {
   public Identifier userId() {
     T message = tail();
     return switch (message.role()) {
-      case ASSISTANT, SYSTEM -> message.recipientId();
+      case ASSISTANT -> message.recipientId();
       case USER -> message.senderId();
+      case SYSTEM -> throw new IllegalStateException(
+          "ThreadState should never hold a SYSTEM message");
     };
   }
 
   public Identifier botId() {
     T message = tail();
     return switch (message.role()) {
-      case ASSISTANT, SYSTEM -> message.senderId();
+      case ASSISTANT -> message.senderId();
       case USER -> message.recipientId();
+      case SYSTEM -> throw new IllegalStateException(
+          "ThreadState should never hold a SYSTEM message");
```

Comment:
nit: This message is used in several places. Declaring it as a string constant should make it easy to modify and more consistent.